src/plugins/discord/fetcher.test.js

```diff
@@ -0,0 +1,268 @@
+      },
+      nick: "nickname",
+      roles: ["test role"],
+    });
+
+    const options = () => ({
+      membersLimit: 2,
+      messagesLimit: 100,
+      reactionsLimit: 100,
+    });
+
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, options());
+      await fetcher.members("1", "0");
+      expect(fetch.mock.calls[0]).toEqual([
+        "/guilds/1/members?after=0&limit=2",
+      ]);
+    });
+
+    it("handles response", async () => {
+      const response = [testMember(1)];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, options());
+      const {results} = await fetcher.members("1", "0");
+      expect(results).toEqual(response);
+    });
+
+    it("returns correct endCursor", async () => {
+      const response = [testMember(1), testMember(2)];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, options());
+      const {endCursor} = (await fetcher.members("1", "0")).pageInfo;
+      expect(endCursor).toEqual(2);
+    });
+
+    describe("next page", () => {
+      const testHasNextPage = async (nextPage) => {
+        const response = [testMember(1)];
+        const fetch = jest.fn(() => Promise.resolve(response));
+        const fetcher = new Fetcher(fetch, {
+          membersLimit: nextPage ? 1 : 2,
+          messagesLimit: 100,
+          reactionsLimit: 100,
+        });
+        const {hasNextPage} = (await fetcher.members("1", "0")).pageInfo;
+        return hasNextPage;
+      };
+
+      it("next page = true", async () => {
+        expect(await testHasNextPage(true)).toBe(true);
+      });
+
+      it("next page = false", async () => {
+        expect(await testHasNextPage(false)).toBe(false);
+      });
+    });
+  });
+
+  describe("fetch reactions", () => {
+    const emoji = {id: "1", name: "emojiname"};
+
+    const options = () => ({
+      membersLimit: 100,
+      messagesLimit: 100,
+      reactionsLimit: 2,
+    });
+
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, options());
+      await fetcher.reactions("1", "2", emoji, "0");
+      expect(fetch.mock.calls[0]).toEqual([
+        `/channels/1/messages/2/reactions/emojiname:1?after=0&limit=2`,
+      ]);
+    });
+
+    it("handles response", async () => {
+      const response = [{id: 123, emoji}];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, options());
+      const {results} = await fetcher.reactions("3", "2", emoji, "0");
+      const expected = [{emoji, channelId: "3", messageId: "2", authorId: 123}];
+      expect(results).toEqual(expected);
+    });
+
+    it("returns correct endCursor", async () => {
+      const response = [
+        {id: 1, emoji},
+        {id: 2, emoji},
+      ];
```

Comment:
Missing type: `const response: Emoji[]`

Note: I think this would show a bug, because I think there's a missing spread operator.